subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy

```diff
@@ -231,6 +231,28 @@ class TestTaskIntegrationTest extends AbstractIntegrationSpec {
         result.assertOutputContains("The setTestClassesDir(File) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use the setTestClassesDirs(FileCollection) method instead.")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/3627")
+    def "can reference properties from TestTaskReports when using @CompileStatic"() {
```

Comment:
Does this test fail without your change?

Reply:
Very good point, and it does fail without the change.